src/Evolution/Systems/ScalarAdvection/Subcell/NeighborPackagedData.hpp

```diff
@@ -0,0 +1,276 @@
+
+        for (const auto& mortar_id : mortars_to_reconstruct_to) {
+          if constexpr (Dim == 1) {
+            const Direction<Dim>& direction = mortar_id.first;
+            Index<Dim> extents = subcell_mesh.extents();
+
+            // Note : 1D mortar has only one face point
+            const size_t num_face_pts = 1;
+            vars_on_face.initialize(num_face_pts);
+
+            using velocity_field = ScalarAdvection::Tags::VelocityField<Dim>;
+            const auto& velocity_on_face =
+                db::get<evolution::dg::subcell::Tags::OnSubcellFaces<
+                    velocity_field, Dim>>(box);
+            data_on_slice(make_not_null(&get<velocity_field>(vars_on_face)),
+                          gsl::at(velocity_on_face, direction.dimension()),
+                          extents, direction.dimension(),
+                          direction.side() == Side::Lower
+                              ? 0
+                              : extents[direction.dimension()] - 1);
+
+            // Reconstruct field variables on faces
+            call_with_dynamic_type<void,
+                                   typename ScalarAdvection::fd::Reconstructor<
+                                       Dim>::creatable_classes>(
+                &recons, [&element, &mortar_id, &neighbor_subcell_data,
+                          &subcell_mesh, &vars_on_face,
+                          &volume_vars_subcell](const auto& reconstructor) {
+                  reconstructor->reconstruct_fd_neighbor(
+                      make_not_null(&vars_on_face), volume_vars_subcell,
+                      element, neighbor_subcell_data, subcell_mesh,
+                      mortar_id.first);
+                });
+
+            // Compute fluxes
+            ScalarAdvection::subcell::compute_fluxes<Dim>(
+                make_not_null(&vars_on_face));
+
+            tnsr::i<DataVector, Dim, Frame::Inertial> normal_covector = get<
+                evolution::dg::Tags::NormalCovector<Dim>>(
+                *db::get<evolution::dg::Tags::NormalCovectorAndMagnitude<Dim>>(
+                     box)
+                     .at(mortar_id.first));
+            for (auto& t : normal_covector) {
+              t *= -1.0;
+            }
+
+            // Compute the packaged data
+            packaged_data.initialize(num_face_pts);
+            evolution::dg::Actions::detail::dg_package_data<System<Dim>>(
+                make_not_null(&packaged_data),
+                dynamic_cast<const derived_correction&>(boundary_correction),
+                vars_on_face, normal_covector, {std::nullopt}, box,
+                typename derived_correction::dg_package_data_volume_tags{},
+                dg_package_data_argument_tags{});
+
+            neighbor_package_data[mortar_id] = std::vector<double>{
+                packaged_data.data(),
+                packaged_data.data() + packaged_data.size()};
+
+          } else {
```

Comment:
There's a lot of duplicated code between the 1d and 2d cases. Why not share all that? It seems like there's really only a couple lines that differ.